doc/source/tutorial/stats/sampling.rst

```diff
@@ -0,0 +1,327 @@
+.. _non-uniform-random-number-sampling:
+
+=====================================================
+Universal Non-Uniform Random Number Sampling in SciPy
+=====================================================
+
+.. currentmodule:: scipy.stats
+
+SciPy provides an interface to many universal non-uniform random number
+samplers to sample random variates from a wide variety of univariate
+continuous and discrete distributions. Implementations of a fast C library
+called `UNU.RAN <http://statmath.wu.ac.at/software/unuran/>`__ are used
+for speed and performance. Please look at
+`UNU.RAN's documentation <http://statmath.wu.ac.at/software/unuran/doc/unuran.html>`__
+for an in-depth explanation of these methods. It is heavily referred
+for writing this tutorial and the documentation of all the samplers.
+
+
+Introduction
+------------
+
+Random variate generation is the small field of research that deals with
+algorithms to generate random variates from various distributions. It is
+common to assume that a uniform random number generator is available.
+This is a program that produces a sequence of independent and identically
+distributed continuous U(0,1) random variates (i.e. uniform random variates
+on the interval (0,1)). Of course, real-world computers can never generate
+ideal random numbers and they cannot produce numbers of arbitrary precision
+but state-of-the-art uniform random number generators come close to this
+aim. Thus random variate generation deals with the problem of transforming
+such a sequence of U(0,1) random numbers into non-uniform random variates.
+
+Some methods to do that are:
+
+* The Inversion method: When the inverse :math:`F^{-1}` of the cumulative
+  distribution function is known, then random variate generation is easy.
+  We just generate a uniformly U(0,1) distributed random number U and
+  return :math:`X = F^{-1}(U)`. See
```

Comment:
Let's add something like: As closed form solutions for the inverse are rarely available, one usually needs to rely on approximations of the inverse. The followings approaches are implemented in SciPy:

Reply:
Done.